### DISCUSSION TITLE
Bucket CRD in Flux source controller

### DISCUSSION THREAD

**Discussion by TechGirl-3** [fluxcd/flux2#1238] [Q&A]
Hi,

I am trying to include bucket crd in flux source controller as mentioned in the document, however I am getting following error:-

no matches for kind "Bucket" in version "source.toolkit.fluxcd.io/v1beta1"
Document Link: [https://toolkit.fluxcd.io/components/source/buckets/]

Need suggestion how to integrate bucket crd in flux.

Thanks in Advance.

**Comment by stefanprodan** [comment]
Install the latest flux release that contains the Bucket CRD https://github.com/fluxcd/flux2/releases

**Reply by TechGirl-3** [reply]
Hi,

I have installed latest version for source and notification controller but still I get below error:-
error: unable to recognize "../../templates/gotk-components.yaml": no matches for kind "Bucket" in version "source.toolkit.fluxcd.io/v1beta1"

flux-source-controller - v0.11.0
flux-notification-controller - v0.12.0

I also updated the manifest file and included following for bucket resource:-
apiVersion: source.toolkit.fluxcd.io/v1beta1
kind: Bucket
metadata:
  name: ${BUCKET_NAME}
  namespace: default
spec:
  interval: 5m
  provider: aws
  bucketName: ${BUCKET_NAME}
  endpoint: s3.amazonaws.com
  region: us-east-1
  timeout: 30s

**Reply by stefanprodan** [reply]
> I have installed latest version for source and notification controller

The CRDs are separated from the controllers. Please uninstall flux, delete gotk-components.yaml and use `flux bootstrap git` to setup correctly Flux.

**Reply by TechGirl-3** [reply]
Hi @stefanprodan 

I did flux bootstrap and and got the new yaml files. I am applying these files using kubectl apply command. I also added the Bucket CRD resource in gotk-sync.yaml file. It gives following error:-

Error from server (AlreadyExists): error when creating "../../templates/gotk-components.yaml": customresourcedefinitions.apiextensions.k8s.io "buckets.source.toolkit.fluxcd.io" already exists
